### DISCUSSION TITLE
Return type of formState { errors }

### DISCUSSION THREAD

**Discussion by mirik999** [react-hook-form/react-hook-form#5014] [Q&A]
Hi, i'm using react-hook-form with typescript. I'm trying to add dynamic inputs with Controller, everything is okey except error handling. My form state type is 

```Javascript
  type FormValues = {
    subCategory: {
      parentId: string;
      name: string;
      tabName: string;
    }[];
  };
```

errors log output is ( without subCategory )
```Javascript
  {
     name: { message: '....', type .... },
     tabName: { message: '....', type .... },
  }
```

But when i'm trying to get errors.name or tabName i'm getting a type error - TS2339: Property 'tabName' does not exist on type 'DeepMap '.  DeepMap contains FormValues. 
Validation realized with yup.
Maybe i'm doing something wrong.

**Comment by leapful** [comment]
@mirik999 

Please provide the codesandbox to get more detail about the issue. As I can see in your comment, your type "FormValues" is an array type but it seems the "errors" object is current handling object type.

**Reply by mirik999** [reply]
https://codesandbox.io/s/gracious-cdn-bt0tj?file=/src/Subcategories.tsx
